app/controllers/api/graphql_controller.rb

```
@@ -0,0 +1,78 @@
+module Api
+  class GraphqlController < ActionController::Base
+    force_ssl :if => :require_ssl?
+    rescue_from Exception, :with => :generic_exception if Rails.env.production?
```

If you raise exceptions in application logic (and this is Ruby, you surely do), this might not be a good idea. GraphQL APIs are generally expected to always return 200 and instead populate an `errors` attribute. I would try and leave as much deviation of errors codes in the actual transport logic (HTTP) and abide by the convention of responding 200 on expected errors.

---

I don't think rescuing from `Exception` is a good idea. This will catch **all** exceptions and `generic_exception` will respond with `500` which isn't correct for all situations as @ohadlevy mentioned in https://github.com/theforeman/foreman/pull/5336/files#r176930235 

Also related here is https://github.com/theforeman/foreman/pull/5336/files#r176930474 as we could raise a RecordNotFound exception here with `find` which we can hook into and return the correct graphql error object and an accompanying `200` status (corresponding to what @chrisarcand mentioned above).

There must be a better solution for catching unwanted exceptions.